pkg/redirectpolicy/manager.go

```diff
@@ -668,42 +679,81 @@ func (rpm *Manager) upsertConfigWithNamedPorts(config *LRPConfig, pods ...*podMe
 				}
 			}
 			if len(bes4) > 0 {
-				rpm.upsertServiceWithBackends(config, feM, pod.id, bes4)
+				rpm.updateFrontendMapping(config, feM, pod.id, bes4)
+				upsertFes = append(upsertFes, feM)
 			} else if len(bes6) > 0 {
-				rpm.upsertServiceWithBackends(config, feM, pod.id, bes6)
+				rpm.updateFrontendMapping(config, feM, pod.id, bes6)
+				upsertFes = append(upsertFes, feM)
```

Comment:
This still adds the same `feM` to `upsertFes` once per pod backend right?

Reply:
Good catch! Fixed.